### ISSUE TITLE
Duplicate declarations should cause syntax error

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/espree#196]
The following code:
```js
let a = 20;
const a = 22;
```
I currently parsed successfully, but should produce syntax error per @michaelficarra

[15.1.1](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-scripts-static-semantics-early-errors):

> * It is a Syntax Error if the LexicallyDeclaredNames of ScriptBody contains any duplicate entries.
> * It is a Syntax Error if any element of the LexicallyDeclaredNames of ScriptBody also occurs in the VarDeclaredNames of ScriptBody.

[15.2.1.1](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-module-semantics-static-semantics-early-errors):

> * It is a Syntax Error if the LexicallyDeclaredNames of ModuleItemList contains any duplicate entries.
> * It is a Syntax Error if any element of the LexicallyDeclaredNames of ModuleItemList also occurs in the VarDeclaredNames of ModuleItemList.

[13.2.1](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-block-static-semantics-early-errors):

> * It is a Syntax Error if the LexicallyDeclaredNames of StatementList contains any duplicate entries.
> * It is a Syntax Error if any element of the LexicallyDeclaredNames of StatementList also occurs in the VarDeclaredNames of StatementList.

... etc.

**Comment by nzakas** [comment]
Technically you are correct, but I'm not sure it's with the effort to implement such a check.

**Comment by ilyavolodin** [comment]
Up to you. I just wanted to record this somewhere so we could at least have a record of it. I verified that Esprima also doesn't flag this as syntax error. Couldn't find online tester for acorn (or found one, but it was broken).

**Comment by nzakas** [comment]
Also happens in Acorn. Going to close this as "won't fix" unless it becomes a serious issue.
